### DISCUSSION TITLE
[Question][Hw5] About cmp

### DISCUSSION THREAD

**Discussion by mrpositron** [prosyslab-classroom/cs348-information-security#605] [Q&A]
Could you provide the information about what `cmp` should do?

**Comment by mrpositron** [comment]
I saw previous posts on `cmp`. However, they do not explain what should it do.

**Comment by RiceBiscuits** [comment]
A concrete version of `cmp` should verify that provided predicate is true or false, and return a result.
An abstract version of `cmp` should verify that provided predicate, and return an abstracted result that subsumed all possible cases.
